rational/src/lib.rs

```diff
@@ -668,6 +668,164 @@ impl RatioErrorKind {
     }
 }
 
+impl FromPrimitive for Ratio<BigInt> {
+    fn from_i64(n: i64) -> Option<Self> {
+        Some(Ratio::from_integer(n.into()))
+    }
+
+    fn from_u64(n: u64) -> Option<Self> {
+        Some(Ratio::from_integer(n.into()))
+    }
+
+    fn from_f32(n: f32) -> Option<Self> {
+        Ratio::from_float(n)
+    }
+
+    fn from_f64(n: f64) -> Option<Self> {
+        Ratio::from_float(n)
+    }
+}
+
+macro_rules! from_primitive_integer {
+    ($typ:ty) => {
+        impl FromPrimitive for Ratio<$typ> {
+            fn from_i64(n: i64) -> Option<Self> {
+                <$typ as NumCast>::from(n).map(|n| Ratio::new(n, One::one()))
+            }
+
+            fn from_u64(n: u64) -> Option<Self> {
+                <$typ as NumCast>::from(n).map(|n| Ratio::new(n, One::one()))
+            }
+
+            fn from_f32(n: f32) -> Option<Self> {
+                Self::approximate_float(n)
+            }
+
+            fn from_f64(n: f64) -> Option<Self> {
+                Self::approximate_float(n)
+            }
+        }
+    }
+}
+
+from_primitive_integer!(i8);
+from_primitive_integer!(i16);
+from_primitive_integer!(i32);
+from_primitive_integer!(i64);
+
+impl<T: Integer + Signed + Bounded + NumCast + Copy> Ratio<T> {
```

Comment:
Clone instead of Copy would be possible, yes. The others are all needed unless you can think of another way for the conversions between T and F and back